app/lib/lets-encrypt.server.ts

```diff
@@ -29,8 +87,93 @@ class LetsEncrypt {
       directoryUrl: this.#directoryUrl,
       accountKey: this.#accountKey,
     });
+
+    await this.#registerReloadAccount();
+
+    return this;
+  };
+
+  /**
+   * This fn extracts and requests challenge information from / based on
+   * the authorizations array that we got in the order.
+   */
+  #extractChallenges = async () => {
+    this.#authorizations = await this.#client!.getAuthorizations(this.#order!);
+
+    /**
+     * Each authorization (a soon to be common name or alt name
+     * we requested in createOrder => identifiers) contains one or
+     * more challenges. At least one challenge per authorization needs
+     * to be passed for the authorization to be granted
+     *
+     * Later, we can use the granted authorizations in our certificate
+     */
+    const promises = this.#authorizations.map(async (authorization) => {
+      const selectedChallenge = authorization.challenges.find(
+        ({ type }) => type === 'dns-01'
+      ) as AcmeDnsChallenge;
+
+      // Get key for challenge
+      const keyAuthorization = await this.#client!.getChallengeKeyAuthorization(selectedChallenge!);
```

Comment:
Instead of `selectedChallenge!`, can we throw earlier if it's really missing?  I realize it's not likely, but given that it's possible, from the point of view of the types, we should deal with it.

Reply:
You are right. Updated!